scico/linop/abel.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2022 by SCICO Developers
+# All rights reserved. BSD 3-clause License.
+# This file is part of the SCICO package. Details of the copyright and
+# user license can be found in the 'LICENSE' file distributed with the
+# package.
+
+"""Abel transform LinearOperator wrapping the pyabel package.
+
+Abel transform LinearOperator wrapping the
+`pyabel <https://github.com/PyAbel/PyAbel>`_ package.
+"""
+
+import math
+from typing import Optional
+
+import numpy as np
+
+import jax
+import jax.numpy as jnp
+import jax.numpy.fft as jnfft
+
+import abel
+
+from scico.linop import LinearOperator
+from scico.typing import Array, JaxArray, Shape
+from scipy.linalg import solve_triangular
+
+
+class AbelProjector(LinearOperator):
+    r"""Abel transform projector based on `PyAbel <https://github.com/PyAbel/PyAbel>`_.
+
+    Perform Abel transform (parallel beam tomographic projection of
+    cylindrically symmetric objects) for a 2D image. The input 2D image
+    is assumed to be centered and left-right symmetric.
+    """
+
+    def __init__(self, img_shape: Shape):
+        """
+        Args:
+            img_shape: Shape of the input image.
+        """
+        self.proj_mat_quad = _pyabel_daun_get_proj_matrix(img_shape)
+
+        super().__init__(
+            input_shape=img_shape,
+            output_shape=img_shape,
+            input_dtype=np.float32,
+            output_dtype=np.float32,
+            adj_fn=self._adj,
+            jit=True,
+        )
+
+    def _eval(self, x: JaxArray) -> JaxArray:
```

Comment:
Do we need this much indirection (`_eval` calls `_proj` calls `_pyabel_transform`)? I believe the structure was mainly used elsewhere for purposed of host callback, but I don't think that's used here.

Reply:
I agree. It probably does not require this much indirection.

Reply:
Fixed it.